TACC/Core-CMS-Custom
# Hotfix/update reg form for form validation

## Overview

This addresses the issue on prod that displays the pattern attribute for a telephone number input for an added contact

## Related

- [WP-232](https://jira.tacc.utexas.edu/browse/WP-232)
- [WP-233](https://jira.tacc.utexas.edu/browse/WP-233)
- [WP-234](https://jira.tacc.utexas.edu/browse/WP-234)
- mimics #195

## Changes

The checkForBlankInputs javascript was changed using whitelist methods to allow characters in input. At the same time, the script blacklists some explicit scripting character patterns. The innerHTML was causing some of the regrex escape characters to not be read correctly. Adjusted the use of the script within the add contact button's script.


## Testing

1. Go to [http://localhost:8000/register/request-to-submit/](http://localhost:8000/register/request-to-submit/)
2. Fill out form and add a contact
3. Try to input **<script> </script> javascript.exe, anything in "quotes" or 'quotes" { hello.exe } thisshouldwork@test.com into any field** into a contact name field and make sure invalid characters are removed
